metagraph/src/graph/representation/succinct/boss_chunk_construct.cpp

```diff
@@ -489,39 +525,25 @@ generate_dummy_1_kmers(size_t k,
         }
     }
 
+    // remove all unmerged chunks
+    common::remove_chunks(real_F_W);
+
     uint64_t num_sink = 0;
     uint64_t num_source = 0;
     for (TAlphabet i = 0; i < alphabet_size; ++i) {
+        real_chunks[i].finish();
```

Comment:
The ordering here is a bit confusing to me. Since `finish()` is not called until after `remove_chunks`, isn't there a chance that a chunk may be removed before it's merged?

Reply:
all chunks have been read by this point, so we can safely remove them. `finish()` is called to flush the output buffers, when no more reading is done at this point.